Backpack/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,263 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import UIKit
+
+public class BPKRating: UIView {
+
+    // MARK: - Public properties
+    public var value: Float {
+        didSet { updateLookAndFeel() }
+    }
+    public var ratingScale: BPKRatingScale {
+        didSet { updateLookAndFeel() }
+    }
+    public var size: BPKRatingSize {
+        didSet { updateLookAndFeel() }
+    }
+
+    public var showScale: Bool {
+        didSet { updateLookAndFeel() }
+    }
+    public var titleView: UIView {
+        didSet { updateLookAndFeel() }
+    }
+
+    public var title: String {
+        get {
+            titleLabel.text ?? ""
+        }
+        set {
+            titleLabel.text = newValue
+            titleView = titleLabel
+            updateLookAndFeel()
+        }
+    }
+    public var subtitle: String? {
+        get {
+            subtitleLabel.text ?? ""
+        }
+        set {
+            subtitleLabel.text = newValue
+            updateLookAndFeel()
+        }
+    }
+
+    // MARK: - Subviews
+    private let horizontalStackView: UIStackView = {
+        let horizontalStack = UIStackView()
+        horizontalStack.axis = .horizontal
+        horizontalStack.spacing = BPKSpacingMd
+        horizontalStack.alignment = .lastBaseline
+        return horizontalStack
+    }()
+    private let ratingValueAndScaleStackView: UIStackView = {
+        let horizontalStack = UIStackView()
+        horizontalStack.axis = .horizontal
+        horizontalStack.alignment = .lastBaseline
+        return horizontalStack
+    }()
+    private let titleSubtitleStackView: UIStackView = {
+        let verticalStack = UIStackView()
+        verticalStack.axis = .vertical
+        verticalStack.alignment = .leading
+        verticalStack.distribution = .fill
+        return verticalStack
+    }()
+
+    private let ratingValueLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textLabel1)
+        label.textColor = BPKColor.textPrimaryColor
+        return label
+    }()
+    private let ratingScaleLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textCaption)
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    private let titleLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textLabel1)
+        label.textColor = BPKColor.textPrimaryColor
+        return label
+    }()
+    private let subtitleLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textBodyDefault)
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
```

Comment:
nit: these all need to be `private lazy var`

Reply:
I prefer to not use `lazy var` here because: 
- using `let` show that it is a constant. 
- All 4 labels are created by `init`'s call to `setup()`, using lazy doesn't delay the initialization

Reply:
Makes sense